### ISSUE TITLE
1.12 Stone ring lost markings

### ISSUE THREAD

**Issue by Knito58** [RootsTeam/Roots#241]
MC 1.12
Forge 14.21.1.2413
Roots 2 0.025

Built my base near a stone ring which had the markings. After some minecraft days (and some crashes, too) the ring has lost its marks. Another ring which is 400m away still has all of the markings and converted stone into runestone successfully.

Is there any chance that the marks of the base ring come back?

**Comment by elucent** [comment]
...huh. I'm not sure. Are there any blocks above you by any chance? The markings appear on the top blocks of their column.

It's possible that you may have removed the entity that represents the Moonbeam Circle through a /kill command or something, or perhaps it despawned somehow even though it's not supposed to.

I'm not sure what would cause a bug like this, but if I find anything I'll let you know.

**Comment by Knito58** [comment]
No, there's nothing visible above. I cannot verify truly if there are any invisible blocks as I cannot fly atm. From the bottom view the sky is clear. Built a dirt pillar to y level 180 - nothing. The entities seem to be gone for good. Cannot /kill or anything as cheats are off. Gamemode is SSP "normal".

**Comment by Knito58** [comment]
I have issues when returning from Aroma1997's Mining Dimension to the Overworld. Maybe that is where the tile entities of the ring have gone to.

https://bitbucket.org/Aroma1997/miningworld/issues/126/hoppers-stuck-when-exiting-mining-world

I could not trace down the cause for this error until today. It seems to me that the re-loading of the world when re-entering the Overworld from the mining world is damaged somehow. In the process of the reload all entities of the starting chunks seem to get lost in SSP. Maybe its not a problem on Multiplayer but I cannot verify this.

Lost entities in my case: Chickens, DaVincis Vessel, the Thuttech Elevator in example. 

That would explain to me why the markings of the base ring have gone. I now believe strongly that it is not a fault of Roots2 that the markings are now lost. 

But aside of this a new question:

Would it be possible to re-introduce the lost entities into an old stone ring?
Or even create new rings?

**Comment by elucent** [comment]
Not at the moment. I had plans to implement a means of creating magical points in creative but never got to it.

**Comment by Knito58** [comment]
I thought about some magic which could be done when the player has advanced in the mod. The generation procedure could be bound to a expensive and hard to make device which would be consumed in the process. So the player could put his own ring on the roof-garden of his building.

("the player" = Konitor)

:)

**Comment by Knito58** [comment]
Seems now as if it not was the Aroma Mining World as the issue also found place when leaving the nether. Maybe one of the other 116 mods might cause problems when a world gets initialized.

Lost entities when coming from mining portal: https://www.dropbox.com/s/r23ayjsmoj7f5vn/Konitor3logs.zip?dl=1

Lost entities when coming from nether portal: https://www.dropbox.com/s/ksbafylppeso0zb/Konitor4logs.zip?dl=1

**Comment by elucent** [comment]
Hmmm. I might migrate these over to some new system in the future but no promises. I'm not sure what's causing this bug in the meantime though.

**Comment by Knito58** [comment]
I believe I've found the reason for why the markings - along with alot of other entities including vessels, lifts and panels - did disappear.

It was a Forge setting which I did change for 1.12: "Remove erroring entities". And of course I forgot about it. Sorry for that. I just found the switch when migrating to 1.12.1 and this time did not change it. After some days of play I can say that the default setting does not kill entities.

Entities may start "erroring" from a Forge's view when a chunk gets reloaded and the entities in the loaded chunk do not reply quickly enough to the question which Forge asks each entity after loading:

"R u ok?" 

But then Forge does not wait long enough for the the answer. The still sleepy (not correctly initialized) entity only opens one eye but cannot speak yet. Forge thinks "U r notok." and deletes it.

So there's hope that only few people (like me) derped with this setting.

The error also happened when returning to the base after a long voyage. I had not necessarily to leave the world and re-enter it. Re-loading a chunk could also lead to a loss of entities.
